### DISCUSSION TITLE
How to change protobuf-java dependency to protobuf-javalite?

### DISCUSSION THREAD

**Discussion by stosabon** [trustwallet/wallet-core#2353] [Q&A] [answered]
I have duplicating classes because I use javalite version in my project. Is there any way to change this dependency?

**Comment by hewigovens** [comment]
lite version might not work, could you please try full protobuf version?

**Reply by stosabon** [reply]
I have a conflict because another library uses lite version.
And almost all android libraries use lite version because it is recommended

**Reply by vcoolish** [reply]
possible to exclude lite version from your lib, so it will use normal one?

**Reply by hewigovens** [reply]
We're using 'io.grpc:grpc-protobuf' and I tried lite version, it doesn't work, we need full version to support reflection etc

**Reply by stosabon** [reply]
Firebase library can't be used with normal version. It requires lite :) Do you have any plans for migration maybe? Seems that  in Android it is recommended to use lite everywhere.

**Reply by stosabon** [reply]
@hewigovens @vcoolish Do you use firebase-analytics with javalite version? If yes, how did you solve this issue?
How can I change protobuf-java to protobuf-javalite in your library? Am I allowed to do it? What part of code will break if we change it to javalite?

**Reply by hewigovens** [reply]
We don't use firebase

**Comment by hewigovens** [comment]
can you try this https://github.com/trustwallet/wallet-core/discussions/2402#discussioncomment-3230204?

**Reply by stosabon** [reply]
It is not working since I need not only perf from firebase. If I exclude dependencies from firebase, my project doesn't build

**Reply by stosabon** [reply]
Please try to refactor wallet-core to javalite version. It is a big problem to integrate it to big projects, because google force to use javalite in Android

**Comment by hewigovens** [comment, marked as answer]
We're testing this: https://github.com/trustwallet/wallet-core/pull/2410
